### PR TITLE
Correctly save transitive dependencies

### PR DESCRIPTION
**Pull request by iarna**
Direct deps get `package.json` and `package-lock.json`.  Transitive deps get `package-lock.json`.

This is a first step to ditching the depth=0 on `npm update`.  The other thing we need to do before we do that is fix a bug where asking to update a named transitive dependency can cause an infinite loop in `outdated`.

### CODE REVIEW

File: lib/install.js [hunk 1]
```diff
@@ -215,23 +217,31 @@ function Installer (where, dryrun, args) {
   this.progress = {}
   this.noPackageJsonOk = !!args.length
   this.topLevelLifecycles = !args.length
-  this.dev = npm.config.get('dev') || (!/^prod(uction)?$/.test(npm.config.get('only')) && !npm.config.get('production')) || /^dev(elopment)?$/.test(npm.config.get('only'))
-  this.prod = !/^dev(elopment)?$/.test(npm.config.get('only'))
-  this.rollback = npm.config.get('rollback')
-  this.link = npm.config.get('link')
-  this.global = this.where === path.resolve(npm.globalDir, '..')
+  this.dev = opts.dev != null ? opts.dev : npm.config.get('dev') || (!/^prod(uction)?$/.test(npm.config.get('only')) && !npm.config.get('production')) || /^dev(elopment)?$/.test(npm.config.get('only'))
```
Comment (zkat) [inline lib/install.js]: Can I ask that we parenthesize, and maybe split between lines, all these different conditionals? I can never tell how it's actually supposed to parse when you do ternary + or + and all in on line >_<

Comment (zkat) [reply lib/install.js]: this change otherwise looks good

**Review by zkat — APPROVED** [review]
LGTM -- just a small readability change that you can do if you want.

File: lib/install.js (hunk 1 above)
Comment (iarna) [reply lib/install.js]: er wow, yeah, that looks awful…

**Pull request merged**